### PR TITLE
Angular. NgRx

### PR DESCRIPTION
**Pull request by aleksander1802**
1. Task: [Link](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/angular/NgRX.md)
2. Screenshot: localhost_4200_youtube (2)
3. Deploy: [Link](https://aleksander1802.github.io/angular2023Q4/)
4. Done: 20.11.2023 / Deadline: 21.11.2023
5. Score: 100 / 100

# Scoring: Angular. NgRx

- [x] _Admin Page_ is generated (**+5**)
- [x] _Favorite Page_ is generated (**+5**)
- [x] NgRx package is used and storage is created (**+5**)
- [x] _Custom Cards_ are saved in the store (**+10**)
- [x] Videos (with _favorite button_) from the YouTube API received via Effects are saved in store (**+20**)
- [x] _Custom Cards_ (without _favorite button_) are displayed on _List Page_ combined with YouTube videos (**+20**)
- [x] _Favorite Page_ displays all marked videos from _List Page_ or _View Page_ (**+10**)
- [x] Clicking the _favorite button_ on the card on _Favorite Page_ removes item from the store and from the page
      immediately (**+15**)
- [x] _List Page_ displays 20 items with pagination and _Custom Cards_ are added to the beginning of the list on 1
      page (**+10**)

### CODE REVIEW

File: src/app/favorite/components/favorite-item/favorite-item.component.html
Comment (ShkrobNastya) [inline src/app/favorite/components/favorite-item/favorite-item.component.html:1]: this component looks similar to app-search-item. Can we reuse app-search-item for favourite page?

File: src/app/youtube/components/pagination/pagination.component.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    nextPage() {
+        this.currentPage += 1;
+        localStorage.setItem('currentPage', this.currentPage.toString());
+        this.store.dispatch(VideoCardActions.getNextPage());
+    }
+
+    prevPage() {
+        if (this.currentPage > 1) {
+            this.currentPage -= 1;
+            localStorage.setItem('currentPage', this.currentPage.toString());
```
Comment (ShkrobNastya) [inline src/app/youtube/components/pagination/pagination.component.ts:48]: **you don't need to fix it** 
in future would be better to save current page in url params, not in localStorage (if you want to save page view after pagereload)

File: src/app/store/effects/video-cards.effects.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+        ))
+    ));
+
+    getPrevPage$ = createEffect(() => this.actions$.pipe(
+        ofType(VideoCardActions.getPrevPage),
+        switchMap(() => this.resultsService.getPrevPageResults().pipe(
+            map((videoCard) => VideoCardActions.getPrevPageSuccess({ videoCard })),
+            catchError((error) => of(VideoCardActions.getVideoCardFailure({ error })))
+        ))
```
Comment (ShkrobNastya) [inline src/app/store/effects/video-cards.effects.ts:40]: **you don't need to fix it**
as I see, getNextPage, getPrevPage looks similar: effects, reducers
I think the logic can be combined

**Review by ShkrobNastya — CHANGES_REQUESTED** [review]

File: src/app/favorite/components/favorite-item/favorite-item.component.html
Comment (aleksander1802) [reply src/app/favorite/components/favorite-item/favorite-item.component.html:1]: You're right, there is a similarity between `app-favorite-item` and `app-search-item`. To promote code reuse and maintainability, I'll explore the possibility of refactoring and reusing `app-search-item` in the `FavoriteModule`. I'll make sure to maintain the necessary functionality while consolidating common components. Thanks again for pointing this out!

File: src/app/youtube/components/pagination/pagination.component.ts (hunk 1 above)
Comment (aleksander1802) [reply src/app/youtube/components/pagination/pagination.component.ts:48]: It's a great idea, I agree 100% that it's much more convenient this way.

**Review by ShkrobNastya — APPROVED** [review]

File: src/app/favorite/components/favorite-item/favorite-item.component.html
Comment (aleksander1802) [reply src/app/favorite/components/favorite-item/favorite-item.component.html:1]: The fix is ready. Made the item-search component global. So I'm reusing it in Search Results and in Favorite Results. Thus, the unused favorite item component had to be deleted.

**Pull request merged**